### DISCUSSION TITLE
nuget.config packageSourceCredentials support

### DISCUSSION THREAD

**Discussion by theKBro** [renovatebot/renovate#30498] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab with ghcr.io/renovatebot/renovate:37.440.7

### Please tell us more about your question or problem

I would like to use renovate to update packages from private nuget repositories.  
The nuget.config file contains all information needed to access the private packageSources  
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="myPrivateRepo" value="https://gitlab.myCompany.com/api/v4/groups/94/-/packages/nuget/index.json" />
  </packageSources>
  <packageSourceCredentials>
    <myPrivateRepo>
      <add key="Username" value="shared_read" />
      <add key="ClearTextPassword" value="passwordInClearText" />
    </myPrivateRepo>
  </packageSourceCredentials>
</configuration>
```

but renovate does not seem to use the `packageSourceCredentials` at all.  
We have many different repositories with different read access tokens. Is there any way that this is working or do I need to add `hostRules` to my runner configuration for all possible autodetected repositories? 

### Logs (if relevant)

<details><summary>Logs</summary>

```
DEBUG: Found NuGet.config at nuget.config (repository=myRepo)
DEBUG: Adding registry URL https://gitlab.myCompany.com/api/v4/groups/94/-/packages/nuget/index.json (repository=myRepo)
       "registryUrl": "https://gitlab.myCompany.com/api/v4/groups/94/-/packages/nuget/index.json",
       "sourceMappedPackagePatterns": undefined

```

```
DEBUG: GET https://gitlab.myCompany.com/api/v4/groups/94/-/packages/nuget/metadata/otherProject/index.json = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=401 retryCount=0, duration=33) (repository=myRepo)
DEBUG: Datasource unauthorized (repository=myRepo)
       "datasource": "nuget",
       "packageName": "otherProject",
       "url": "https://gitlab.myCompany.com/api/v4/groups/94/-/packages/nuget/metadata/otherProject/index.json"
```

</details>

**Comment by rarkins** [comment]
This requires a feature request:
- nuget manager to extract clear text credentials during extract phase
- nuget manager to append to these credentials if relevant instead of overwriting it in the artifacts update section

Please create a reproduction repo with the above which can be forked and debugged. Naturally the URLs and credentials can be dummy ones but they should be as close to real examples as possible

**Reply by theKBro** [reply]
I tried my best to provide a reproduction repo with
https://github.com/theKBro/dummyApp

the package is from https://github.com/theKBro/dummyDependency

hopefully this helps
